packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,274 @@
+   */
+  export function hasPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+
+    return new Promise<boolean>((resolve) => {
+      resolve(sendAndHandleSdkError('permissions.has', permissions));
+    });
+  }
+
+  /**
+   * Requests user permission for visualMedia
+   * @beta
+   * @returns Promise that will resolve with true if the user consented permission for media, or with false otherwise,
+   * In case of an error, promise will reject with the error. Function can also throw a NOT_SUPPORTED_ON_PLATFORM error
+   */
+  export function requestPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+
+    return new Promise<boolean>((resolve) => {
+      resolve(sendAndHandleSdkError('permissions.request', permissions));
+    });
+  }
+
+  /**
+   * Checks if visualMedia capability is supported by the host
+   * @returns boolean to represent whether media is supported
+   * @beta
+   * @throws Error if {@linkcode app.initialize} has not successfully completed
+   */
+  export function isSupported(): boolean {
+    return ensureInitialized(runtime) && runtime.supports.visualMedia && runtime.supports.permissions ? true : false;
+  }
+
+  /**
+   * To enable this image capability will let the app developer ask the user to get images from camera/local storage
+   * @beta
+   */
+  export namespace image {
+    /**
+     * Capture one or multiple image(s) throughing camera.
+     * @beta
+     * @param imageInputs - The input params to customize the image(s) to be captured
+     * @returns Promise that will resolve with {@link VisualMediaFile[]} object or reject with an error.
+     */
+    export function captureImages(imageInputs: ImageProperties): Promise<VisualMediaFile[]> {
+      ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      if (!imageInputs || imageInputs == null || imageInputs.visualMediaCount > 10) {
+        throw new Error('Must supply the valid image(s)');
+      }
+      // waiting and return the response from hub-SDK
+      return sendAndHandleSdkError<VisualMediaFile[]>('visualMedia.image.captureImages', imageInputs);
+    }
+
+    /**
+     * Upload the existing image(s) from camera roll to the mos app.
+     * @beta
+     * @param imageInputs - The input params to customize the image(s) to be captured
+     * @returns Promise that will resolve with {@link VisualMediaFile[]} object or reject with an error.
+     */
+    export function uploadImages(imageInputs: ImageProperties): Promise<VisualMediaFile[]> {
```

Comment:
>uploadImages [](http://example.com/codeflow?start=20&length=12)

I'm a little confused about the intended usage here.

Are you intending that there be two different functions, one for a developer to call if they want photos from the camera and one of they want photos from the gallery? If so, it seems problematic that both of these functions accept an `ImageProperties` parameter that lets them pick that. What would it mean if a developer called `captureImages` but specified a source of `Gallery`?

Reply:
Apologies for any confusion. I'd greatly appreciate any suggestions you might have.
When a developer invokes captureImage() with the Gallery as the source, an error will be returned. 
Both uploadImages() and captureImages() accept the ImageProperties parameter. The primary distinction between them is the source: either the camera or the gallery. For the host app, this source determines whether the camera or gallery is launched, achieved by calling two distinct UI packages. The result will be a list of image files (VisualMediaFile). 
To ensure the API remains clear and user-friendly for developers, I introduced uploadImages and captureImages. In our current media API, we use selectMedia() for capturing and uploading images, videos, and audio. I'm also considering introducing a selectImages() function that can manage both capturing and uploading image behaviors. The host apps can then decide which UI package to invoke based on the source. However, at this stage, we can't intercept if the user provides an incorrect source.

Reply:
That's a great suggestion, updated!